Storage/tests/Unit/SigningHelperTest.php

```diff
@@ -789,6 +790,52 @@ private function mockConnection($credentials = null, $rw = null)
 
         return $conn->reveal();
     }
+
+    public function testRetrySignBlobSuccessFirstAttempt()
+    {
+        $signBlobFn = function () {
+            return 'signature';
+        };
+
+        $res = $this->helper->proxyPrivateMethodCall('retrySignBlob', [
+            $signBlobFn
+        ]);
+
+        $this->assertEquals('signature', $res);
+    }
+
+    public function testRetrySignBlobSuccessAfterRetries()
+    {
+        $attempt = 0;
+        $signBlobFn = function () use (&$attempt) {
+            if (++$attempt < 3) {
+                throw new ServiceException('Transient error', 503);
+            }
+            return 'signature';
+        };
+
+        $res = $this->helper->proxyPrivateMethodCall('retrySignBlob', [
+            $signBlobFn
+        ]);
+
+        $this->assertEquals('signature', $res);
```

Comment:
add attempt assertion
```suggestion
        $this->assertEquals('signature', $res);
        $this->assertEquals(3, $attempt);
```

Reply:
Added